inotify.go

```diff
@@ -101,9 +125,14 @@ func (w *Watcher) Add(name string) error {
 
 // Remove stops watching the the named file or directory (non-recursively).
 func (w *Watcher) Remove(name string) error {
-	name = filepath.Clean(name)
 	w.mu.Lock()
 	defer w.mu.Unlock()
+	return w.remove(name)
+}
+
+// Only call this when w.mu is locked, like from Remove or Close.
```

Comment:
How about a seperate method to return a slice of watches, or a method to simply remove all watches?

Since the watcher is closed when Closing anyway, no new ones will be added, and we could do more fine-grained locking in remove.